salt/log/setup.py

```diff
@@ -905,6 +952,63 @@ def setup_multiprocessing_logging_listener(opts, queue=None):
     __MP_LOGGING_LISTENER_CONFIGURED = True
 
 
+def setup_multiprocessing_zmq_logging(port=None):
+    from salt.utils.platform import is_windows
+
+    global __MP_LOGGING_CONFIGURED
+    global __MP_LOGGING_ZMQ_HANDLER
+
+    if __MP_IN_MAINPROCESS is True and not is_windows():
+        # We're in the MainProcess, return! No multiprocessing logging setup shall happen
+        # Windows is the exception where we want to set up multiprocessing
+        # logging in the MainProcess.
+        return
+
+    try:
+        logging._acquireLock()  # pylint: disable=protected-access
+
+        if __MP_LOGGING_CONFIGURED is True:
+            return
+
+        # Let's set it to true as fast as possible
+        __MP_LOGGING_CONFIGURED = True
+
+        if os.getpid() in zmq_handlers:
+        #if __MP_LOGGING_ZMQ_HANDLER is not None:
+            print("SKIP LOGGIN BECAUSE HANDLER EXISTS %d" %(os.getpid()))
+            return
+
+        # The temp null and temp queue logging handlers will store messages.
+        # Since noone will process them, memory usage will grow. If they
+        # exist, remove them.
+        __remove_null_logging_handler()
+        __remove_queue_logging_handler()
+
+        # Let's add a queue handler to the logging root handlers
+        #print("SET UP LOGGING %d %d" % ( os.getpid(), port or get_multiprocessing_logging_port()))
+        handler = ZMQHandler(port or get_multiprocessing_logging_port())
+#        zmq_handlers[os.getpid()] = handler
+        __MP_LOGGING_ZMQ_HANDLER = handler
+        logging.root.addHandler(handler)
```

Comment:
The is probably what we did with the old MP logging, but in a child process we should probably clear any handlers and just have the zmq handler...

Reply:
I've fixed this.